src/foam/core/JSON.js

```diff
@@ -186,6 +186,12 @@ foam.CLASS({
       help: 'If true, Property objects are passed as __Property__ references rather than by value.',
       value: true
     },
+    {
+      class: 'Boolean',
+      name: 'passClassesByReference',
+      help: 'If true, Class objects are passed as __Class__ references rather than by value.',
```

Comment:
This isn't really an accurate description and I don't think we ever will need to change this flag.  There's no way to pass a class object by value, as classes are not serializable.  They are always passed by name/id, but its just a question of whether we use the long form `{ "class": "__Class__",  "name": "Foo" }` or the short form `"Foo"`

This isn't a decision that the JSONOutputter needs tomake or even be aware of.  In the Class property toJSON, we should always just output the id, same as we do for Dates, we always just output the raw date value directly rather than the long form `{ "class":"__Timestamp__"...` version.

See here: https://github.com/foam-framework/foam2/blob/master/src/foam/core/types.js#L87 in either case we are returning a simple value.  For dates we have extra config on whether we want to format as a number or a string, but for Class references we don't need to make this distinction.